internal/engine/compiler/impl_amd64_test.go

```diff
@@ -391,6 +391,42 @@ func TestAmd64Compiler_readInstructionAddress(t *testing.T) {
 	})
 }
 
+func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
+	env := newCompilerEnvironment()
+	compiler := env.requireNewCompiler(t, newAmd64Compiler, nil).(*amd64Compiler)
+
+	rv := []*runtimeValueLocation{{register: amd64.RegAX}, {register: amd64.RegCX}, {register: amd64.RegDX}}
+
+	currentRegisters := func() []asm.Register {
+		out := make([]asm.Register, len(rv))
+		for i := range rv {
+			out[i] = rv[i].register
+		}
+		return out
+	}
+
+	tests := []struct {
+		desired, expected []asm.Register
+		swaps             []int
+	}{
+		{desired: []asm.Register{amd64.RegDX, amd64.RegCX, amd64.RegAX}, expected: []asm.Register{amd64.RegDX, amd64.RegCX, amd64.RegAX}, swaps: []int{0, 2}},
```

Comment:
Could you break the lines?

```suggestion
		{
			desired: []asm.Register{amd64.RegDX, amd64.RegCX, amd64.RegAX}, 
			expected: []asm.Register{amd64.RegDX, amd64.RegCX, amd64.RegAX}, 
			swaps: []int{0, 2},
		},
```